fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneScanAllEntriesTest.java

```diff
@@ -0,0 +1,220 @@
+                .setIsGrouped(isGrouped)
+                .setIsSynthetic(isSynthetic)
+                .setPrimaryKeySegmentIndexEnabled(true)
+                .setPartitionHighWatermark(10)
+                .build();
+
+        final long start = Instant.now().toEpochMilli();
+        try (FDBRecordContext context = openContext()) {
+            dataModel.saveRecords(500, start, context, 2);
+            commit(context);
+        }
+        // Scan properties with a limit of 100
+        final ScanProperties scanProperties = new ScanProperties(ExecuteProperties.newBuilder()
+                .setReturnedRowLimit(100)
+                .setIsolationLevel(IsolationLevel.SERIALIZABLE)
+                .build());
+
+        // This test only uses the match-all query
+        LuceneQueryClause search = LuceneQuerySearchClause.MATCH_ALL_DOCS_QUERY;
+
+        try (FDBDatabaseRunner runner = fdb.newRunner()) {
+            // Create a cursor that reads through all continuations
+            RecordCursor<IndexEntry> cursor = new AutoContinuingCursor<>(
+                    runner,
+                    (context, continuation) -> {
+                        final FDBRecordStore store = dataModel.createOrOpenRecordStore(context);
+                        LuceneScanBounds scanBounds = isGrouped
+                                                      ? LuceneIndexTestValidator.groupedSortedTextSearch(store, dataModel.index, search, null, 2)
+                                                      : LuceneIndexTestUtils.fullTextSearch(store, dataModel.index, search, false);
+                        return store.scanIndex(dataModel.index, scanBounds, continuation, scanProperties);
+                    });
+            // Run the scan with the given query and assert the results
+            assertIndexEntryPrimaryKeys(l -> keyFor(2, 1001L + l), 500, cursor);
+        }
+    }
+
+
+    @Nonnull
+    private static Tuple syntheticKeyFor(final int group, final long parentDocId, final long childDocId) {
+        return Tuple.from(-1, Tuple.from(group, parentDocId), Tuple.from(group, childDocId));
+    }
+
+    @Nonnull
+    private static Tuple keyFor(final int group, final long docId) {
+        return Tuple.from(group, docId);
+    }
+
+    private void assertIndexEntryPrimaryKeyTuples(Set<Tuple> primaryKeys, RecordCursor<IndexEntry> cursor) {
+        List<IndexEntry> indexEntries = cursor.asList().join();
+        assertEquals(primaryKeys,
+                indexEntries.stream().map(IndexEntry::getPrimaryKey).collect(Collectors.toSet()));
+    }
+
+    private void assertIndexEntryPrimaryKeys(Function<Long, Tuple> expectedKey, long expectedSize, RecordCursor<IndexEntry> cursor) {
+        List<IndexEntry> indexEntries = cursor.asList().join();
+        for (long l = 0; l < indexEntries.size(); l++) {
+            assertEquals(expectedKey.apply(l), indexEntries.get((int)l).getPrimaryKey());
```

Comment:
The ordering got me wondering, do we want to test updates?
I have some unrelated scratch work to support updating records, perhaps after that is in, we can add some tests here that update some of the records. I don't think it should break anything, but it could change the ordering.

Reply:
Sure. Not sure what the ordering of docs would be but testing updates should be interesting